### PR TITLE
Feat/bem 2 database setup

### PR DESCRIPTION
**Pull request by heorhii-ms**
[Get knowledge of working with PostgreSQL database and Prisma. Get basic knowledge of working with PostgreSQL, communications and using Prisma.](https://www.notion.so/darly-solutions/Get-knowledge-of-working-with-PostgreSQL-database-and-Prisma-Get-basic-knowledge-of-working-with-Po-c458ffe2a9094420a1d6064f808b8b7f?pvs=4)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,22 +9,28 @@
...
     "start:debug": "nest start --debug --watch",
     "start:prod": "node dist/main",
     "lint": "eslint \"{src,apps,libs,test}/**/*.ts\" --fix",
     "test": "jest",
     "test:watch": "jest --watch",
     "test:cov": "jest --coverage",
     "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
-    "test:e2e": "jest --config ./test/jest-e2e.json"
+    "test:e2e": "jest --config ./test/jest-e2e.json",
+    "docker:run-dev": "docker-compose --env-file .env.dev -f docker-compose-dev.yml up -d",
```
Comment (skarm61) [inline package.json]: docker-compose по умолчанию подтягивает .env файл, так что можно переименовать .env.dev в просто .env и убрать отсюда

File: prisma/migrations/migration_lock.toml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Please do not edit this file manually
+# It should be added in your version-control system (i.e. Git)
+provider = "postgresql"
```
Comment (skarm61) [inline prisma/migrations/migration_lock.toml]: please, setup your editor to always add empty line in the end of the file

File: prisma/seed.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+async function main() {
+  const password = await bcrypt.hash(
+    'Test!123',
+    AUTH_CONSTANTS.ROUNDS_OF_GASHING,
+  );
+
+  const user1 = await prisma.user.upsert({
+    where: {email: 'test1@test.test'},
+    update: {},
```
Comment (skarm61) [inline prisma/seed.ts:16]: I guess you don't need it here

File: prisma/seed.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      email: 'test1@test.test',
+      first_name: 'Test1',
+      last_name: 'Testson',
+      password: password,
+    },
+  });
+
+  const user2 = await prisma.user.upsert({
+    where: {email: 'test2@test.test'},
+    update: {},
```
Comment (skarm61) [inline prisma/seed.ts:27]: the same

File: src/entities/user.entity.ts [hunk 5]
```diff
@@ -1,3 +1,25 @@
...
+  id: number;
+
+  @ApiProperty()
+  createdAt: Date;
+
+  @ApiProperty()
+  updatedAt: Date;
+
+  @ApiProperty()
+  first_name: string;
```
Comment (skarm61) [inline src/entities/user.entity.ts]: please, use one style for all columns.
not camelCase and snakeCase together

File: src/modules/user/user.service.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+import {AUTH_CONSTANTS} from 'src/constants/auth-constants';
+import {PrismaService} from '../prisma/prisma.service';
+import {CreateUserDto} from './dto/create-user.dto';
+import {UpdateUserDto} from './dto/update-user.dto';
+
+@Injectable()
+export class UserService {
+  constructor(private prisma: PrismaService) {}
+  public async create(createUserDto: CreateUserDto) {
```
Comment (skarm61) [inline src/modules/user/user.service.ts:14]: please, add empty line above

File: src/modules/user/user.service.ts [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  }
+
+  public async update(id: number, updateUserDto: UpdateUserDto) {
+    if (updateUserDto.password) {
+      const hashedPassword = await bcrypt.hash(
+        updateUserDto.password,
+        AUTH_CONSTANTS.ROUNDS_OF_GASHING,
+      );
+
+      updateUserDto.password = hashedPassword;
```
Comment (skarm61) [inline src/modules/user/user.service.ts]: never mutate incoming arguments!
it's a bad practice

**Review by skarm61 — CHANGES_REQUESTED** [review]

File: prisma/seed.ts (hunk 3 above)
Comment (heorhii-ms) [reply prisma/seed.ts:16]: @skarm61 
It is required
https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#upsert
image
image

Comment (skarm61) [reply prisma/seed.ts:16]: but did you try without it? 
Could it be that only one of them in the same moment required

Comment (heorhii-ms) [reply prisma/seed.ts:16]: @skarm61 
Yes, I tried.
No, both of them required.

**Review by skarm61 — APPROVED** [review]

**Pull request merged**